File: impl/js/src/errors/policyVersionException.ts
/**
 * Thrown by `Policy.from(...)` when loading a policy when a
 * `PolicyDefinition`'s `version` is incompatible with what this
 * implementation supports - a different MAJOR, or a MINOR higher than
 * what's understood within a supported MAJOR.
 * `PATCH` never affects this decision.
 */
export class PolicyVersionException extends Error {
  constructor(message: string) {
    super(message)
    this.name = "PolicyVersionException"
  }
}

File: impl/java/src/main/java/com/cptnfizzbin/keycard/errors/PolicyVersionException.java
package com.cptnfizzbin.keycard.errors;

import lombok.experimental.StandardException;

/**
 * Thrown by {@code Policy.from(...)} when loading a policy when a
 * PolicyDefinition's {@code version} is incompatible with what this
 * implementation supports - a different MAJOR, or a MINOR higher than
 * what's understood within a supported MAJOR.
 * {@code PATCH} never affects this decision.
 */
@StandardException
public class PolicyVersionException extends RuntimeException {
}
